spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppUsedInStreamException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.controller;
+
+/**
+ * @author Christian Tzolov
+ */
+public class AppUsedInStreamException extends RuntimeException {
```

Comment:
The name of this exception doesn't quite correspond to the message.  `ApplicationNotFoundException` would be better, though it feels like we have too many fine grained exceptions, maybe look to reuse another one since nothing is catching this specific exception.

Reply:
Turns out we already have a `NoSuchAppException` that is identical to this one, will use that instead.